### PR TITLE
Fix all maybe-uninitialized warnings

### PR DESCRIPTION
**Pull request by derobins**

### CODE REVIEW

File: hdf/util/he_main.c [hunk 1]
```diff
@@ -664,8 +664,7 @@ putElement(char *file, uint16 tag, uint16 ref, char *data, int32 len)
 int
 writeGrp(char *file)
 {
-    int    i;
-    uint16 ref;
+    uint16 ref = DFREF_NONE;
```
Comment (derobins) [inline hdf/util/he_main.c:667]: Not initialized (DFREF_NONE is the best "bad" value)

File: mfhdf/dumper/hdp_list.c [hunk 2]
```diff
@@ -892,39 +892,41 @@ do_list(intn curr_arg, intn argc, char *argv[], intn help)
     int32       an_id     = FAIL; /* annotation interface handle */
     intn        ret_value = SUCCEED;
 
+    /* Do this early, to avoid uninitialized warnings in cleanup code */
+    init_list_opts(&list_opts);
+
```
Comment (derobins) [inline mfhdf/dumper/hdp_list.c:897]: This was initialized after some `goto done` calls, so the compiler complains about the string in list_opts not always being initialized.

File: mfhdf/hdfimport/hdfimport.c [hunk 3]
```diff
@@ -2501,7 +2501,7 @@ pixrep(struct Input *in, struct Raster *im)
 static int32
 create_SDS(int32 sd_id, int32 nt, struct Input *in)
 {
-    int32 sds_id;
+    int32 sds_id = FAIL;
```
Comment (derobins) [inline mfhdf/hdfimport/hdfimport.c:2504]: Uninitialized. FAIL is the "bad" value.

File: mfhdf/hdfimport/hdfimport.c [hunk 4]
```diff
@@ -2712,7 +2712,8 @@ process(struct Options *opt)
     int32          len;
     FILE          *strm = NULL;
     int32          hdf;
-    int32          sd_id, sds_id;
+    int32          sd_id  = FAIL;
+    int32          sds_id = FAIL;
```
Comment (derobins) [inline mfhdf/hdfimport/hdfimport.c:2716]: Uninitialized. FAIL is the "bad" value.

File: mfhdf/test/tcoordvar.c [hunk 5]
```diff
@@ -88,7 +88,8 @@ test_dim1_SDS1(void)
     float32        sds1_data[] = {0.1, 2.3, 4.5, 6.7, 8.9};
     float32        out_data[5];
     int32          dimsize[1];
-    int32          sds_id, file_id, dim_id, index;
+    int32          sds_id, file_id, dim_id;
+    int32          index = FAIL;
```
Comment (derobins) [inline mfhdf/test/tcoordvar.c:92]: Uninitialized. FAIL is the "bad" value.

**Review by byrnHDF — APPROVED** [review]

**Pull request merged**